### ISSUE TITLE
SyntaxError when importing in browser via ES6 modules

### ISSUE THREAD

**Issue by bradberger** [Flagsmith/flagsmith-js-client#106]
We're using ES6 modules to build an application. The application JavaScript is complied with Rollup and imported as a ES6 module. For example (where index.js is our application code):

```html
<body>
  <script type="module" src="index.js"></script>
</body>
```

Our application (which is compiled to index.js in the example above) imports the Flagsmith JavaScript SDK via NPM (shortened for clarity):

```javascript
import flagsmith from 'flagsmith';

flagsmith.init();
```

This approach looks fine in the IDE, but in the browser throws an error:

`Uncaught SyntaxError: The requested module './../node_modules/flagsmith/index.js' does not provide an export named 'default'`

Using this approach, it also appears that `window.flagsmith` is defined. We're using ES6 modules to avoid "polluting" the global `window` object, so this is undesired behavior. 

The use case is developing a JS SDK ourselves, and want to use our own instance of Flagsmith to control functionality. That SDK is used inside of third-party sites so we can't set anything on the `window` object.

**Comment by kyle-ssg** [comment]
Hmm I see. I think it's quite common for modules to have separate bundles for es6, e.g. lodash-es.

Let me know but it seems the solution could be to have flagsmith/es and flagsmith/react-es, also these bundles could look to avoid using window.flagsmith. Does that sound suitable @bradberger ?

**Comment by bradberger** [comment]
@kyle-ssg That'd be perfect!

**Comment by kyle-ssg** [comment]
Having said this, It's a bit strange since the nextjs/react examples import flagsmith fine https://github.com/Flagsmith/flagsmith-js-client/blob/main/examples/react/src/index.tsx. Is there a valid reason for this do you think?

**Comment by bradberger** [comment]
Hmm good point. We've used it in React, too, without issue. I'm wondering if React compiles to something less that ES6 for compatibility. Maybe we need to do that too. I can check into that, it might be a possible solution. But all the same ES6 support would be great if it's possible.

**Comment by kyle-ssg** [comment]
Yeah I bet there's some clever webpack config for create-react-app / nextjs in general. Looking into this now 👍

**Comment by kyle-ssg** [comment]
Could you try npm i flagsmith-es as a replacement to flagsmith @bradberger ?

**Comment by bradberger** [comment]
@kyle-ssg That did the trick! It still appears to populate the `window.flagsmith` variable, but importing `createFlagsmithInstance()` works perfectly and the global instance isn't our instance.

**Comment by kyle-ssg** [comment]
Ok let me just sort the global!
